pkg/storage/segments/store.go

```diff
@@ -217,6 +218,31 @@ func (s *segmentStore) Get(ctx context.Context, path storj.Path) (
 			return nil, Meta{}, err
 		}
 
+		// get the total non nil nodes
+		totalNonNilNodes := 0
+		for _, v := range nodes {
+			if v != nil {
+				totalNonNilNodes++
+			}
+		}
+
+		// calculate how many minimum nodes needed based on t = k + (n-o)k/o
+		t := es.RequiredCount() + ((es.TotalCount()-totalNonNilNodes)*es.RequiredCount())/totalNonNilNodes
+
+		minNumOfNodes := 0
+		for i, v := range nodes {
+			if v != nil {
+				minNumOfNodes++
+				if minNumOfNodes >= t {
+					minNumOfNodes = i
+					break
+				}
+			}
+		}
+
+		nodes = nodes[:minNumOfNodes]
```

Comment:
In case of nil nodes, you might end up with more nil nodes and never being able to download the piece.